opentelemetry-sdk/src/opentelemetry/sdk/logs/__init__.py

```diff
@@ -0,0 +1,174 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import atexit
+from typing import Any, Optional
+
+from opentelemetry.sdk.logs.severity import SeverityNumber
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.util.instrumentation import InstrumentationInfo
+from opentelemetry.trace.span import TraceFlags
+from opentelemetry.util.types import Attributes
+
+
+class LogRecord:
+    """A LogRecord instance represents an event being logged.
+
+    LogRecord instances are created and emitted via `LogEmitter`
+    every time something is logged. They contain all the information
+    pertinent to the event being logged.
+    """
+
+    def __init__(
+        self,
+        timestamp: Optional[int] = None,
+        trace_id: Optional[int] = None,
+        span_id: Optional[int] = None,
+        trace_flags: Optional[TraceFlags] = None,
+        severity_text: Optional[str] = None,
+        severity_number: Optional[SeverityNumber] = None,
+        name: Optional[str] = None,
+        body: Optional[Any] = None,
+        resource: Optional[Resource] = None,
+        attributes: Optional[Attributes] = None,
+    ):
+        self.timestamp = timestamp
+        self.trace_id = trace_id
+        self.span_id = span_id
+        self.trace_flags = trace_flags
+        self.severity_text = severity_text
+        self.severity_number = severity_number
+        self.name = name
+        self.body = body
+        self.resource = resource
+        self.attributes = attributes
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, LogRecord):
+            return NotImplemented
+        return self.__dict__ == other.__dict__
+
+
+class LogData:
+    """Readable LogRecord data plus associated InstrumentationLibrary."""
+
+    def __init__(
+        self,
+        log_record: LogRecord,
+        instrumentation_info: InstrumentationInfo,
+    ):
+        self.log_record = log_record
+        self.instrumentation_info = instrumentation_info
+
+
+class LogProcessor(abc.ABC):
+    """Interface to hook the log record emitting action.
+
+    Log processors can be registered directly using
+    :func:`LogEmitterProvider.add_log_processor` and they are invoked
+    in the same order as they were registered.
+    """
+
+    @abc.abstractmethod
+    def emit(self, log_data: LogData):
```

Comment:
It would be nice if the SpanProcessor would also have a method called `emit` for consistency. No action required from this comment, just a thought.

Reply:
Hmm, what would `emit` mean in the context of span?